dkms.in

```diff
@@ -953,6 +953,13 @@ prepare_signing()
         return
     fi
 
+    if [[ -f "${kernel_config}" ]]; then
+        sign_hash="$(grep "^CONFIG_MODULE_SIG_HASH=" "${kernel_config}" | sed 's/CONFIG_MODULE_SIG_HASH=//;s/"//g')"
+        sign_hash="${sign_hash:-sha512}"
```

Comment:
This piece is very odd - there is no single default sighash in the kernel. The default is picked based on what's built-in.

As such, I cannot see how the CONFIG option can be missing. Do you mind adding a note where you've seen this - "noticed with $distro $version, where $sighash is used but the CONFIG entry is missing".

That said, do you mind if we put your (good PRs) on hold for a day or two - would be nice to land https://github.com/dell/dkms/pull/262 without forcing the developer to rebase yet again.

Reply:
If the kernel lacks sig_hash option/support, perhaps we should skip the test all together?

Reply:
> This piece is very odd - there is no single default sighash in the kernel. The default is picked based on what's built-in.
> As such, I cannot see how the CONFIG option can be missing. 

The kernel may be built without the module signing feature completely. `# CONFIG_MODULE_SIG_HASH is not set` in kconfig.

> Do you mind adding a note where you've seen this - "noticed with $distro $version, where $sighash is used but the CONFIG entry is missing".

OK

> That said, do you mind if we put your (good PRs) on hold for a day or two - would be nice to land #262 without forcing the developer to rebase yet again.

I don't mind. 

> If the kernel lacks sig_hash option/support, perhaps we should skip the test all together?

Currently dkms always signs the built modules, and there is no problem using signed modules on systems without module signing feature. The test is checking whether the hash algorithm of module signature is correct, not whether the system supports module signing. The test is necessary IMO.

Reply:
Thanks for bearing with my questions :bowing_man: